### PR TITLE
Added discord.py min. version checking

### PR DESCRIPTION
**Pull request by CrafterSvK**
Resolves: #80
Decorator version function checks minimum version of discord.py and replies to user if command is not available.

### CODE REVIEW

File: core/acl.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
+from functools import wraps
+from typing import Optional
+
+import discord
 from discord.ext import commands
+from discord.ext.commands import Context
```
Comment (sinus-x) [inline core/acl.py]: This is unnecessary, just juse `commands.Context`

File: core/lang/cs.ini [hunk 2]
```diff
@@ -8,3 +8,6 @@ no named subcommand = Příkaz **((name))** nemá podpříkaz **((subcommand))**
 
 [create_embed]
 footer = Odpověď pro
+
+[version]
+reply = Příkaz není dostupný
```
Comment (sinus-x) [inline core/lang/cs.ini:13]: I think we should say something a bit different:
```diff
- reply = Příkaz není dostupný
+ reply = Příkaz není v této verzi dostupný
+ reply = Příkaz není v této verzi discord.py dostupný
+ reply = To teď neumím
```
or something else, feel free to improvise.

File: core/lang/en.ini [hunk 3]
```diff
@@ -8,3 +8,6 @@ no named subcommand = Command **((name))** does not have subcommand **((subcomma
 
 [create_embed]
 footer = Answer for
+
+[version]
+reply = This command is not available
```
Comment (sinus-x) [inline core/lang/en.ini:13]: See comment for `cs.ini`

File: core/acl.py [hunk 4]
```diff
@@ -113,3 +119,34 @@ async def repeat(self, ctx, *, input: str):
...
+                elif version_info.minor > minor:
+                    version_check = False
+            elif version_info.major > major:
+                version_check = False
+
+            if not version_check:
+                return await func(*args, **kwargs)
+
+            if isinstance(args[1], Context):
+                await args[1].message.reply(tr("version", "reply"))
```
Comment (sinus-x) [inline core/acl.py]: Now that I think about it sometimes we may want to return plain `False` with no message. Can you add named argument to the check? `@check.version(2, 0, reply=False)`/`@check.version(2, 0, reply=True)`.

File: core/acl.py [hunk 5]
```diff
@@ -113,3 +119,34 @@ async def repeat(self, ctx, *, input: str):
 
     # user's roles are not mapped to any ACL group, return default
     return rule.default
+
+
+def version(major: int, minor: int, micro: Optional[int] = None):
+    """Decorator that specifies minimum discord.py version"""
```
Comment (sinus-x) [inline core/acl.py]: `"""Specify minimal discord.py version"""` maybe?

Also, please add short `param` and `return` descriptions, just as `check.acl()` has.

File: core/acl.py [hunk 6]
```diff
@@ -113,3 +119,34 @@ async def repeat(self, ctx, *, input: str):
 
     # user's roles are not mapped to any ACL group, return default
     return rule.default
+
+
+def version(major: int, minor: int, micro: Optional[int] = None):
```
Comment (sinus-x) [inline core/acl.py]: Return type would be nice. Just `) -> bool:`.

File: core/acl.py [hunk 7]
```diff
@@ -113,3 +119,34 @@ async def repeat(self, ctx, *, input: str):
...
+
+    def decorator(func):
+        @wraps(func)
+        async def wrapper(*args, **kwargs):
+            version_info = discord.version_info
+            version_check = True
+
+            if version_info.major == major:
+                if version_info.minor == minor:
+                    if micro is None or version_info.micro >= micro:
```
Comment (sinus-x) [inline core/acl.py]: Shouldn't this be `if micro is not None and version_info.micro >= micro`?

**Review by sinus-x — CHANGES_REQUESTED** [review]
I left a few comments there. It looks nice, I'll merge it when they are resolved.

File: core/acl.py (hunk 7 above)
Comment (CrafterSvK) [reply core/acl.py]: My bad, was thinking about checking for wrong version at first and then I did this version.

File: core/acl.py (hunk 6 above)
Comment (CrafterSvK) [reply core/acl.py]: It does not always return `bool`, function is returned when version is correct

Comment (sinus-x) [reply core/acl.py]: Oh, I see, this is not implemented to be used inside of `commands.check()` decorator. I didn't think of this solution. Well, we'll see how it works, we can always update it if it does not work as intended before the 1.0 release sometime in the future.

In this case the return type should be `Union[bool, Callable]`. Or would returning `None` be better solution to insufficient version?

Comment (CrafterSvK) [reply core/acl.py]: I think returning `None` makes more  sense as it is not a check that something is correct but the calling of the function/method depends on this, returning `None` means that it had nothing to call.

File: core/check.py [hunk 8]
```diff
@@ -121,8 +120,17 @@ async def repeat(self, ctx, *, input: str):
     return rule.default
 
 
-def version(major: int, minor: int, micro: Optional[int] = None):
-    """Decorator that specifies minimum discord.py version"""
+def version(
+    major: int, minor: int, micro: Optional[int] = None, reply=True
```
Comment (sinus-x) [inline core/check.py:124]: `reply` here is missing a type hint

**Review by sinus-x — CHANGES_REQUESTED** [review]
Only one small thing now

**Review by sinus-x — APPROVED** [review]
Looks good now. Thanks

**Pull request merged**